Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs

```diff
@@ -97,49 +97,88 @@ public void OnError(Exception error)
         {
         }
 
+        /// <summary>
+        /// Provides the observer with new data.
+        /// <seealso cref="IObserver{T}.OnNext(T)"/>
+        /// </summary>
+        /// <param name="evnt">The current notification information.</param>
         public void OnNext(KeyValuePair<string, object> evnt)
         {
-            switch (evnt.Key)
+            try
             {
-                case HttpOutStartEventName:
+                switch (evnt.Key)
                 {
-                    this.OnActivityStart((HttpRequestMessage)this.startRequestFetcher.Fetch(evnt.Value));
-                    break;
-                }
+                    case HttpOutStartEventName:
+                        {
+                            var request = this.startRequestFetcher.Fetch(evnt.Value) as HttpRequestMessage;
 
-                case HttpOutStopEventName:
-                {
-                    this.OnActivityStop(
-                        (HttpResponseMessage)this.stopResponseFetcher.Fetch(evnt.Value),
-                        (HttpRequestMessage)this.stopRequestFetcher.Fetch(evnt.Value),
-                        (TaskStatus)this.stopRequestStatusFetcher.Fetch(evnt.Value));
-                    break;
-                }
+                            if (request != null)
+                            {
+                                this.OnActivityStart(request);
+                            }
```

Comment:
Here and in the other places, do we need to do something in "else"?
It looks like a subtle change in the `OnNext()` behavior without it:
- previously, we tried to execute the code and either succeeded or the failure was reported
- currently, we try to execute the code and either succeed or silently continue